src/Aspire.Hosting.AWS/CloudFormation/CloudFormationExtensions.cs

```diff
@@ -0,0 +1,81 @@
+    /// Add a CloudFormation stack for provisioning application resources.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <param name="templatePath">The path to the CloudFormation template that defines the CloudFormation stack.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationResource> AddAWSCloudFormationProvisioning(this IDistributedApplicationBuilder builder, string stackName, string templatePath)
+    {
+        var resource = new CloudFormationResource(stackName, templatePath);
+        var cfBuilder = builder.AddResource(resource);
+
+        builder.Services.AddLifecycleHook<CloudFormationLifeCycle>(sp => ActivatorUtilities.CreateInstance<CloudFormationLifeCycle>(sp, resource));
+        return cfBuilder;
+    }
+
+    /// <summary>
+    /// The AWS credential profile to use for resolving credentials to make AWS service API calls.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="awsSdkConfig">The name of the AWS credential profile.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithAWSSDKReference(this IResourceBuilder<ICloudFormationResource> builder, IResourceBuilder<IAWSSDKConfigResource> awsSdkConfig)
+    {
+        builder.Resource.AWSSDKConfig = awsSdkConfig.Resource;
+        return builder;
+    }
+
+    /// <summary>
+    /// The configured Amazon CloudFormation service client used to make service calls.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="cloudFormationClient">The AWS CloudFormation service client.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithAWSCloudFormationClient(this IResourceBuilder<ICloudFormationResource> builder, IAmazonCloudFormation cloudFormationClient)
+    {
+        builder.Resource.CloudFormationClient = cloudFormationClient;
+        return builder;
+    }
+
+    /// <summary>
+    /// Add a reference of a CloudFormations stack to a project. The output parameters of the CloudFormation stack are added to the project IConfiguration.
+    /// </summary>
+    /// <typeparam name="TDestination"></typeparam>
+    /// <param name="builder"></param>
+    /// <param name="cloudFormationResourceBuilder">The CloudFormation resource.</param>
+    /// <param name="configSection">The config section in IConfiguration to add the output parameters.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<TDestination> WithAWSCloudFormationReference<TDestination>(this IResourceBuilder<TDestination> builder, IResourceBuilder<ICloudFormationResource> cloudFormationResourceBuilder, string configSection = "AWS::Resources")
+        where TDestination : IResourceWithEnvironment
+    {
+        builder.WithEnvironment(context =>
+        {
+            if (context.PublisherName == "manifest" || cloudFormationResourceBuilder.Resource.Outputs == null)
```

Comment:
Is it valid to not have outputs?

Reply:
It is valid but in this use case I would think uncommon. In theory you could have a template that creates some resources that the applications doesn't need to know anything about.